nixos/modules/system/boot/networkd.nix

```diff
@@ -253,22 +464,35 @@ let
 
       sectionFooOverUDP = checkUnitConfig "FooOverUDP" [
         (assertOnlyFields [
-          "Port"
           "Encapsulation"
+          "Port"
+          "PeerPort"
           "Protocol"
+          "Peer"
+          "Local"
         ])
-        (assertPort "Port")
         (assertValueOneOf "Encapsulation" ["FooOverUDP" "GenericUDPEncapsulation"])
+        (assertHasField "Port")
```

Comment:
the port does not have to be set if you're using this as a one-way tunnel.

Reply:
```suggestion
```